lib/api/controllers/authController.js

```diff
@@ -7,26 +7,53 @@ module.exports = function AuthController (kuzzle) {
 
   this.passport = new PassportWrapper(kuzzle);
 
-  this.login = function (requestObject) {
+  this.logout = function (requestObject, context) {
+    var
+      deferred = q.defer(),
+      response;
+
+    kuzzle.pluginsManager.trigger('auth:logout', context)
+      .then(kuzzle.repositories.token.expire(context.token))
+      .then(() => {
+        if (Boolean(context.connection.id)) {
```

Comment:
Should distinguish between REST actions (using `context.connection.type === 'rest'`) and a `logout()` request executed without a `context.connection.id`, which should be a `BadRequestError`.

Moreover, using generic `throws` leads to generic error, and Kuzzle errors are preferred. :-)